pkg/apis/pipeline/v1beta1/task_conversion.go

```diff
@@ -18,12 +18,19 @@ package v1beta1
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
+	"reflect"
+
+	"github.com/tektoncd/pipeline/pkg/apis/version"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	v1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
 	"knative.dev/pkg/apis"
 )
 
+const TaskDeprecationsAnnotationKey = "tekton.dev/v1beta1.task-deprecations"
```

Comment:
I named it as `TaskDeprecationsXXX` because I think all future deprecations should go under this key. Added some comments to clarify the behavior. wdyt?

Reply:
thanks for adding the comments, SGTM